tests/manage/z_cluster/test_flexible_scaling_ui.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+
+from ocs_ci.framework.testlib import (
+    ManageTest,
+    skipif_no_lso,
+    tier1,
+    vsphere_platform_required,
+    skipif_ui_not_support,
+    bugzilla,
+    ui,
+    ignore_leftovers,
+)
+from ocs_ci.ocs.resources.pod import get_osd_pods
+from ocs_ci.ocs.resources.storage_cluster import osd_encryption_verification
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.ui.add_replace_device_ui import AddReplaceDeviceUI
+from ocs_ci.ocs.node import get_nodes, add_disk_to_node
+from ocs_ci.ocs import constants
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.framework import config
+from ocs_ci.ocs.resources.pv import check_available_pvs
+from ocs_ci.ocs.cluster import check_ceph_health_after_add_capacity
+
+
+log = logging.getLogger(__name__)
+
+
+@ui
+@tier1
+@skipif_no_lso
+@ignore_leftovers
+@bugzilla("1943280")
+@vsphere_platform_required
+@skipif_ui_not_support("add_capacity")
+class TestFlexibleScalingUI(ManageTest):
+    """
+    Test Flexible Scaling via UI
+
+    """
+
+    def test_flexible_scaling_ui(self, setup_ui):
+        """
+        Test Procedure:
+        1.Add new disk to worker node
+        2.Add capacity via UI
+        3.Verify all OSD pods move to Running state
+        4.check ceph health after add capacity
+
```

Comment:
I think it will be good to check even the  ceph osd tree and verify that the newly added OSDs are distributed evenly.

Reply:
OSD distribution is having issues when we do in scale here is the bz https://bugzilla.redhat.com/show_bug.cgi?id=2004801